akka-projection-dynamodb/src/main/scala/akka/projection/dynamodb/internal/OffsetStoreDao.scala

```diff
@@ -163,18 +187,31 @@ import software.amazon.awssdk.services.dynamodb.model.WriteRequest
         .returnConsumedCapacity(ReturnConsumedCapacity.TOTAL)
         .build()
 
-      val result = client.batchWriteItem(req).asScala
+      val result = writeWholeBatch(req)
 
       if (log.isDebugEnabled()) {
-        result.foreach { response =>
+        result.foreach { responses =>
           log.debug(
             "Wrote latest timestamps for [{}] slices, consumed [{}] WCU",
             offsetsBatch.size,
-            response.consumedCapacity.iterator.asScala.map(_.capacityUnits.doubleValue()).sum)
+            responses.iterator.flatMap(_.consumedCapacity.iterator.asScala.map(_.capacityUnits.doubleValue)).sum)
         }
       }
       result
-        .map(_ => Done)(ExecutionContext.parasitic)
+        .flatMap { responses =>
+          val lastResponse = responses.last
+          if (lastResponse.hasUnprocessedItems && !lastResponse.unprocessedItems.isEmpty) {
+            val unprocessedSliceItems =
+              lastResponse.unprocessedItems.get(settings.timestampOffsetTable).asScala.toVector
+            val unprocessedSlices = unprocessedSliceItems.map(_.putRequest.item.get(NameSlice).s)
+            log.warn(
+              "Failed to write latest timestamps for [{}] slices: [{}]",
+              unprocessedSlices.size,
+              unprocessedSlices)
```

Comment:
Add a `.mkString(", ")` like the others. These are also the name + slice, but that's fine.